src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -221,18 +231,20 @@ private ImmutableArray<Document> GetPriorityDocuments(Project project)
 
             var seenItems = new HashSet<INavigateToSearchResult>(NavigateToSearchResultComparer.Instance);
             foreach (var projectGroup in orderedProjects)
-                result.AddRange(await Task.WhenAll(projectGroup.Select(p => Task.Run(() => SearchAsync(p, isFullyLoaded, seenItems, cancellationToken)))).ConfigureAwait(false));
+            {
+                var allTasks = projectGroup.Select(p => Task.Run(async () => (p, await SearchAsync(p, isFullyLoaded, seenItems, cancellationToken).ConfigureAwait(false))));
+                result.AddRange(await Task.WhenAll(allTasks).ConfigureAwait(false));
+            }
 
-            return (seenItems.Count, result.ToImmutable());
+            return (foundItems: seenItems.Count > 0, result.ToImmutable());
         }
 
-        private async Task<(Project project, NavigateToSearchLocation location)> SearchAsync(
+        private async Task<NavigateToSearchLocation> SearchAsync(
             Project project, bool isFullyLoaded, HashSet<INavigateToSearchResult> seenItems, CancellationToken cancellationToken)
```

Comment:
there was no need to pass the project in/out.  The caller already has it, so we don't have to give it back to them.